core/kernel.rbs

```diff
@@ -1240,8 +1240,8 @@ module Kernel : BasicObject
   #     0..4
   #     [0..4, 0..4, 0..4]
   #
-  def self?.p: [T] (T arg0) -> T
-             | (untyped, untyped, *untyped) -> Array[untyped]
+  def self?.p: [T < _Inspect] (T arg0) -> T
+             | (_Inspect arg0, _Inspect arg1, *_Inspect rest) -> Array[_Inspect]
```

Comment:
`[T < _Inspect]`? Wynaut

```suggestion
             | [T < _Inspect] (T arg0, T arg1, *T rest) -> Array[T]
```

Reply:
Because that'd would require all the different arguments are the same type then

Reply:
it won’t. It would fall back to `_Inspect`, since `_Inspect` is also `< _Inspect`.